### PR TITLE
Storage: support git + github backed roots

### PR DESCRIPTION
**Pull request by ryantxu**
This PR adds configurable git backends while in dev mode.

If you run in dev mode, it will initialize a config file saved in `${data.dir}/storage/storage.json` -- edit this to point to an appropriate endpoint...

Notable changes:
1. this removes the second service that was used to register HTTP services.  I like this approach better since it keeps more things private
2. This adds a backend for git (and utilities for github)
3. This adds a config file pattern that is saved on disk.... yes it should be in the SQL/system FS, but not quite ready for that yet :)
4. Changes the static security to allow read for admin users
5. Updates the http://localhost:3000/admin/storage so it shows more status info

image

NOTE: this includes changes to the save UX that will be moved to another PR before this is merged

### CODE REVIEW

File: pkg/services/store/service.go [hunk 1]
```diff
@@ -52,6 +54,9 @@ type CreateFolderCmd struct {
 type StorageService interface {
 	registry.BackgroundService
 
+	// Register the HTTP
+	RegisterHTTPRoutes(routing.RouteRegister)
```
Comment (ryantxu) [inline pkg/services/store/service.go:58]: This PR removes the additional HTTP only service, and just adds a single function to register routes.  This lets us keep most of the details private, but still encapsulated in the `http.go` file

File: pkg/services/store/service.go [hunk 2]
```diff
@@ -192,12 +221,21 @@ func ProvideService(sql *sqlstore.SQLStore, features featuremgmt.FeatureToggles,
 				ActionFilesWrite:  allowAllPathFilter,
 				ActionFilesDelete: allowAllPathFilter,
 			}
+
+		// Default ADMIN role is read only
```
Comment (ryantxu) [inline pkg/services/store/service.go:225]: Given that this is grafana admin -- lets default to readonly rather than always hidden

**Review by ArturWierzbicki — APPROVED** [review]
looking good - I feel like we are now badly in need of some integration tests! :)

File: pkg/services/store/storage_git.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+		res.Code = 200
+		res.URL = pr.GetHTMLURL()
+		res.Pending = true
+		res.Hash = *ref.Object.SHA
+		res.Branch = prcmd.headBranch
+		return res, nil
+	}
+
+	// Commit to main
+	if true {
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply pkg/services/store/storage_git.go]: yes

**Pull request merged**